kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -323,6 +323,15 @@ public void close() {
     }
   }
 
+  private void closeEvent(KubernetesClientException cause) {
+    if (forceClosed.get()) {
```

Comment:
One small tweak is to use `getAndSet(true)` to remove the `set(true)` call later.